api/catalog/api/serializers/media_serializers.py

```diff
@@ -73,6 +86,61 @@ class TagSerializer(serializers.Serializer):
     )
 
 
+def get_search_request_source_serializer(media_type):
```

Comment:
I feel like I don't usually see class factories in Python for some reason. Why did you choose this over inheritance? (not suggestion a change, just curious what the thought process is as I almost never reach for class factory in Python but I remember doing it a lot in Scala).

Reply:
It was inheritance-based before but that meant just writing the same thing twice because the fields need to be redefined to change the help text.

Reply:
Oh, I totally glossed over the fact that the help text was changing based on the media type as well :facepalm: Makes a lot of sense!